flag_engine/segments/models.py

```diff
@@ -20,6 +22,12 @@ class SegmentConditionModel:
 
     def matches_trait_value(self, trait_value: typing.Any) -> bool:
         # TODO: move this logic to the evaluator module
+        if (
+            type(trait_value) is str
+            and semver.VersionInfo.isvalid(trait_value)
```

Comment:
Same question as on the flagsmith repo PR - do we need an alternative way of determining if something is a semver?